### PR TITLE
Fix typo in LazySchemaDiffProvider.php

### PR DESCRIPTION
**Pull request by sergiughf**

### CODE REVIEW

**Comment by mikeSimonson** [comment]
@sergiughf Thanks for the fix. Can you patch it everywhere in the code base ?

**Comment by sergiughf** [comment]
Oh man... I don't know how I could push just that. Here's all of it here. Sorry for the brain lag.

File: lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function __construct(LazyLoadingValueHolderFactory $proxyFactory, SchemaD
         $this->originalSchemaManipulator = $originalSchemaManipulator;
     }
 
-    public static function fromDefaultProxyFacyoryConfiguration(SchemaDiffProviderInterface $originalSchemaManipulator)
+    public static function fromDefaultProxyFactoryConfiguration(SchemaDiffProviderInterface $originalSchemaManipulator)
```
Comment (alcaeus) [inline lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php]: Not to be a party pooper, but: is this a BC break?

Comment (stof) [inline lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php]: it actually is. Preserving Bc would be easy, so I suggest doing it: create a method with the old name which would call the new method (and trigger a deprecation notice)

File: lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php [hunk 2]
```diff
@@ -41,6 +41,16 @@ public function __construct(LazyLoadingValueHolderFactory $proxyFactory, SchemaD
 
     public static function fromDefaultProxyFacyoryConfiguration(SchemaDiffProviderInterface $originalSchemaManipulator)
     {
+        trigger_error(
+            sprintf('Function deprecated. Use %s::fromDefaultProxyFactoryConfiguration() instead', self::class),
+            E_USER_DEPRECATED
+        );
+        
+        self::fromDefaultProxyFactoryConfiguration($originalSchemaManipulator);
```
Comment (stof) [inline lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php]: missing `return`

File: lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php [hunk 3]
```diff
@@ -41,6 +41,16 @@ public function __construct(LazyLoadingValueHolderFactory $proxyFactory, SchemaD
 
     public static function fromDefaultProxyFacyoryConfiguration(SchemaDiffProviderInterface $originalSchemaManipulator)
     {
+        trigger_error(
+            sprintf('Function deprecated. Use %s::fromDefaultProxyFactoryConfiguration() instead', self::class),
```
Comment (stof) [inline lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php]: you should mention the deprecation function in the message, to make the warning easier to understand

**Comment by mikeSimonson** [comment]
@sergiughf Thanks

**Comment by sergiughf** [comment]
👍

**Pull request merged**